runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordGameEventType.java

```diff
@@ -24,77 +24,219 @@
+	DUNGEON_QUIDAMORTEM_CAVE("Quidamortem Cave", DiscordAreaType.DUNGEONS, ImmutableList.of(4763)),
+	DUNGEON_RASHILIYIAS_TOMB("Rashiliyta's Tomb", DiscordAreaType.DUNGEONS, ImmutableList.of(11668)),
+	DUNGEON_SARADOMINSHRINE("Saradomin Shrine (Paterdomus)", DiscordAreaType.DUNGEONS, ImmutableList.of(13722)),
+	DUNGEON_SHADE_CATACOMBS("Shade Catacombs", DiscordAreaType.DUNGEONS, ImmutableList.of(13975)),
+	DUNGEON_SHAYZIEN_CRYPTS("Shayzien Crypts", DiscordAreaType.DUNGEONS, ImmutableList.of(6043)),
+	DUNGEON_SMOKE("Smoke Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(12946, 13202)),
+	DUNGEON_SOPHANEM("Sophanem Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(13200)),
+	DUNGEON_STRONGHOLD_SECURITY("Stronghold of Security", DiscordAreaType.DUNGEONS, ImmutableList.of(7505, 8017, 8530, 9297)),
+	DUNGEON_TARNS_LAIR("Tarn's Lair", DiscordAreaType.DUNGEONS, ImmutableList.of(12616, 12615)),
+	DUNGEON_TAVERLEY("Taverley Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(11673, 11672, 11929, 11928, 11417, 11416)),
+	DUNGEON_TEMPLE_OF_IKOV("Temple of Ikov", DiscordAreaType.DUNGEONS, ImmutableList.of(10649, 10905, 10650)),
+	DUNGEON_TEMPLE_OF_MARIMBO("Temple of Marimbo", DiscordAreaType.DUNGEONS, ImmutableList.of(11151)),
+	DUNGEON_THE_WARRENS("The Warrens", DiscordAreaType.DUNGEONS, ImmutableList.of(7070, 7326)),
+	DUNGEON_TOLNA("Dungeon of Tolna", DiscordAreaType.DUNGEONS, ImmutableList.of(13209)),
+	DUNGEON_TOWER_OF_LIFE("Tower of Life Basement", DiscordAreaType.DUNGEONS, ImmutableList.of(12100)),
+	DUNGEON_TUNNEL_OF_CHAOS("Tunnel of Chaos", DiscordAreaType.DUNGEONS, ImmutableList.of(12625)),
+	DUNGEON_UNDERGROUND_PASS("Underground Pass", DiscordAreaType.DUNGEONS, ImmutableList.of(9369, 9370)),
+	DUNGEON_VARROCKSEWERS("Varrock Sewers", DiscordAreaType.DUNGEONS, ImmutableList.of(12954, 13210)),
+	DUNGEON_WATER_RAVINE("Water Ravine", DiscordAreaType.DUNGEONS, ImmutableList.of(13461)),
+	DUNGEON_WATERBIRTH("Waterbirth Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(9886, 10142, 7492, 7748)),
+	DUNGEON_WATERFALL("Waterfall Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(10394)),
+	DUNGEON_WHITE_WOLF_MOUNTAIN_CAVES("White Wolf Mountain Caves", DiscordAreaType.DUNGEONS, ImmutableList.of(11418, 11419, 11675)),
+	DUNGEON_WITCHAVEN_SHRINE("Witchhaven Shrine Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(10903)),
+	DUNGEON_YANILLE_AGILITY("Yanile Agility Dungeon", DiscordAreaType.DUNGEONS, ImmutableList.of(10388));
+
+	private static final Set<Skill> COMBAT_SKILLS = ImmutableSet.of(Skill.ATTACK, Skill.STRENGTH, Skill.DEFENCE, Skill.HITPOINTS, Skill.SLAYER, Skill.RANGED, Skill.MAGIC);
 
 	private final String state;
 	private final String imageKey;
 	private String details;
+
 	private boolean considerDelay = true;
 	private Function<DiscordGameEventType, Boolean> isChanged = (l) -> true;
 	private int priority = 0;
 
+
+	private DiscordAreaType discordAreaType;
+	private List<Integer> regionIds = new ArrayList<>();
```

Comment:
probably dont initialize this when its optional and do early returns in case its null